packages/xdl/src/detach/AndroidShellApp.js

```diff
@@ -1040,24 +1041,83 @@ export async function runShellAppModificationsAsync(
   );
 }
 
-async function buildShellAppAsync(context: StandaloneContext, sdkVersion: string) {
+async function buildShellAppAsync(
+  context: StandaloneContext,
+  sdkVersion: string,
+  buildType: string
+) {
   let shellPath = shellPathForContext(context);
+  const ext = buildType === 'app-bundle' ? 'aab' : 'apk';
 
-  if (context.build.android) {
-    let androidBuildConfiguration = context.build.android;
+  const isRelease = !!context.build.android;
+  // concat on those strings is not very readable, but only alternative here is huge if statement
+  const debugOrRelease = isRelease ? 'Release' : 'Debug';
+  const devOrProd = isRelease ? 'Prod' : 'Dev';
+  const debugOrReleaseL = isRelease ? 'release' : 'debug';
```

Comment:
and we could uppercase them on demand in the code

Reply:
it's quite unreadable as it is, additional logic at this point will make this worse

Reply:
the only clean way I see here is to write all the combination but after adding bundle support this number doubled